c2corg_api/views/document_info.py

```diff
@@ -0,0 +1,105 @@
+from c2corg_api.caching import cache_document_info
+from c2corg_api.models import DBSession
+from c2corg_api.models.cache_version import get_cache_key
+from c2corg_api.models.document import DocumentLocale, Document, \
+    get_available_langs
+from c2corg_api.models.route import Route, RouteLocale
+from c2corg_api.views import etag_cache, \
+    set_best_locale
+from pyramid.httpexceptions import HTTPNotFound
+from sqlalchemy.orm import contains_eager, load_only, joinedload
+from sqlalchemy.orm.util import with_polymorphic
+
+
+class DocumentInfoRest(object):
+    """ Base class for all views that return a basic/info version for a
+    document, that only contains the document_id and title/title_prefix
+    of the requested locale.
+    This view is used by the UI to generate URL slugs.
+
+    If the locale is not available in the requested lang, a locale in the
+    "best" lang is returned.
```

Comment:
OK, so it's the responsibility of the UI to provide at least its interface language if the lang param is not already available in the URL to slugify, right?